ui/core/src/usecases/clp/index.ts

```diff
@@ -16,193 +12,37 @@ export default ({
-
-    return pools[key] ?? null;
-  }
-
-  effect(() => {
-    // When sif address changes syncPools
-    store.wallet.sif.address;
-    syncPools();
-  });
-
-  const actions = {
-    async swap(
-      sentAmount: IAssetAmount,
-      receivedAsset: IAsset,
-      minimumReceived: IAssetAmount,
-    ) {
-      if (!state.address) throw "No from address provided for swap";
-
-      const tx = await services.clp.swap({
-        fromAddress: state.address,
-        sentAmount,
-        receivedAsset,
-        minimumReceived,
       });
 
-      const txStatus = await services.sif.signAndBroadcast(tx.value.msg);
-
-      if (txStatus.state !== "accepted") {
-        // Edge case where we have run out of native balance and need to represent that
-        if (
-          txStatus.code === ErrorCode.TX_FAILED_USER_NOT_ENOUGH_BALANCE &&
-          sentAmount.symbol === "rowan"
-        ) {
-          return reportTransactionError({
-            ...txStatus,
-            code: ErrorCode.TX_FAILED_NOT_ENOUGH_ROWAN_TO_COVER_GAS,
-            memo: getErrorMessage(
-              ErrorCode.TX_FAILED_NOT_ENOUGH_ROWAN_TO_COVER_GAS,
-            ),
-          });
-        }
+      // Then every transaction
 
-        return reportTransactionError(txStatus);
-      }
-
-      return txStatus;
-    },
-
-    async addLiquidity(
-      nativeAssetAmount: IAssetAmount,
-      externalAssetAmount: IAssetAmount,
-    ) {
-      if (!state.address) throw "No from address provided for swap";
-      const hasPool = !!findPool(
-        store.pools,
-        nativeAssetAmount.asset.symbol,
-        externalAssetAmount.asset.symbol,
-      );
-
-      const provideLiquidity = hasPool
-        ? services.clp.addLiquidity
-        : services.clp.createPool;
-
-      const tx = await provideLiquidity({
-        fromAddress: state.address,
-        nativeAssetAmount,
-        externalAssetAmount,
+      services.sif.onNewBlock(async () => {
+        await syncPools();
       });
 
-      const txStatus = await services.sif.signAndBroadcast(tx.value.msg);
-
-      if (txStatus.state !== "accepted") {
-        // Edge case where we have run out of native balance and need to represent that
-        if (txStatus.code === ErrorCode.TX_FAILED_USER_NOT_ENOUGH_BALANCE) {
-          return reportTransactionError({
-            ...txStatus,
-            code: ErrorCode.TX_FAILED_NOT_ENOUGH_ROWAN_TO_COVER_GAS,
-            memo: getErrorMessage(
-              ErrorCode.TX_FAILED_NOT_ENOUGH_ROWAN_TO_COVER_GAS,
-            ),
-          });
-        }
-      }
-      return txStatus;
-    },
-
-    async removeLiquidity(
-      asset: IAsset,
-      wBasisPoints: string,
-      asymmetry: string,
-    ) {
-      const tx = await services.clp.removeLiquidity({
-        fromAddress: state.address,
-        asset,
-        asymmetry,
-        wBasisPoints,
+      effect(() => {
+        // When sif address changes syncPools
+        store.wallet.sif.address;
```

Comment:
This is just a 'read' to signal to the Vue reactive store that we're listening to this field within this effect? A little dicey but I get it.

Reply:
Yeah I hate this. This is Vue reactivity for you from our current implementation. We should consider the store somewhat deprecated and try to fix this up. Not as part of this refactor though. In my head to fix this we aught to probably isolate the store to be it's own service with event handlers.